tasks/util/make_schema.js

```diff
@@ -1,25 +1,24 @@
 var fs = require('fs');
 var path = require('path');
-var jsdom = require('jsdom');
+var JSDOM = require('jsdom').JSDOM;
 
 module.exports = function makeSchema(plotlyPath, schemaPath) {
     return function() {
         var plotlyjsCode = fs.readFileSync(plotlyPath, 'utf-8');
-        jsdom.env({
-            html: '<div></div>',
-            src: [plotlyjsCode],
-            done: function(err, window) {
-                if(err) {
-                    console.log(err);
-                    return;
-                }
 
-                var plotSchema = window.Plotly.PlotSchema.get();
-                var plotSchemaStr = JSON.stringify(plotSchema, null, 4);
-                fs.writeFileSync(schemaPath, plotSchemaStr);
+        var w = new JSDOM('', {runScripts: 'dangerously'}).window;
 
-                console.log('ok ' + path.basename(schemaPath));
-            }
-        });
+        // jsdom by itself doesn't support getContext, and adding the npm canvas
+        // package is annoying and platform-dependent.
+        // see https://github.com/tmpvar/jsdom/issues/1782
+        w.HTMLCanvasElement.prototype.getContext = function() { return null; };
```

Comment:
dunno why it didn't happen with the old version, but with the new jsdom without this patch you get an ugly but nonblocking error `Error: Not implemented: HTMLCanvasElement.prototype.getContext (without installing the canvas npm package)`

Reply:
Yep, no need to add a dependency for this :ok_hand: